qiskit/algorithms/evolvers/real/trotterization/trotter_ops_validator.py

```diff
@@ -0,0 +1,132 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Set of method for validating input to TrotterQrte algorithm."""
+
+import numbers
+from typing import Union
+
+from qiskit.circuit import Parameter, ParameterExpression
+from qiskit.opflow import (
+    SummedOp,
+    PauliOp,
+)
+
+
+def is_op_bound(operator: Union[SummedOp, PauliOp]) -> None:
+    """Checks if an operator provided has all parameters bound.
+
+    Args:
+        operator: Operator to be checked.
+
+    Raises:
+        ValueError: If an operator has unbound parameters.
+    """
+    if len(operator.parameters) > 0:
+        raise ValueError(
+            f"Did not manage to bind all parameters in the Hamiltonian, "
+            f"these parameters encountered: {operator.parameters}."
+        )
+
+
+def validate_hamiltonian_form(hamiltonian: Union[SummedOp, PauliOp]):
+    """Validates that a Hamiltonian is of a correct type and with expected dependence on
+    parameters.
+
+    Args:
+        hamiltonian: Hamiltonian to be validated.
+
+    Raises:
+        ValueError: if an invalid Hamiltonian is provided.
+    """
+    value_error = ValueError(
+        "Hamiltonian term has a coefficient that is not a linear function of a "
+        "single parameter. It is not supported."
+    )
+    if isinstance(hamiltonian, SummedOp):
+        if isinstance(hamiltonian.coeff, ParameterExpression):
+            raise ValueError(
+                f"The coefficient multiplying the whole Hamiltonian cannot be a "
+                f"ParameterExpression. The following coefficient was detected: {hamiltonian.coeff}."
+            )
+        for op in hamiltonian.oplist:
+            if not _is_pauli_lin_single_param(op):
+                raise value_error
+    elif isinstance(hamiltonian, PauliOp):
+        if not _is_pauli_lin_single_param(hamiltonian):
+            raise value_error
+    else:
+        raise ValueError(
+            f"Hamiltonian not a SummedOp/PauliOp which are the only options supported. The "
+            f"following type detected instead: {type(hamiltonian)}."
+        )
+
+
+def _is_pauli_lin_single_param(operator: PauliOp) -> bool:
```

Comment:
This is only used in `validate_hamiltonian_form`, right? Depending on the comment above, we'll have to remember removing this 😄

Reply:
Oh no, it seems that I removed it accidentally when refactoring at some point...